File: 351004/Mukhin/Lab1/restApi/src/main/java/com/example/restApi/services/CommentService.java
package com.example.restApi.services;

import com.example.restApi.dto.request.CommentRequestTo;
import com.example.restApi.dto.response.CommentResponseTo;
import com.example.restApi.exception.NotFoundException;
import com.example.restApi.mapper.CommentMapper;
import com.example.restApi.model.Comment;
import com.example.restApi.repository.CommentRepository;
import org.springframework.stereotype.Service;

import java.time.LocalDateTime;
import java.util.List;
import java.util.stream.Collectors;

@Service
public class CommentService {
    private final CommentRepository commentRepository;
    private final CommentMapper commentMapper;

    public CommentService(CommentRepository commentRepository, CommentMapper commentMapper) {
        this.commentRepository = commentRepository;
        this.commentMapper = commentMapper;
    }

    public List<CommentResponseTo> getAll() {
        return commentRepository.findAll().stream()
                .map(commentMapper::toDto)
                .collect(Collectors.toList());
    }

    public CommentResponseTo getById(Long id) {
        Comment comment = commentRepository.findById(id)
                .orElseThrow(() -> new NotFoundException("Comment not found with id: " + id));
        return commentMapper.toDto(comment);
    }

    public CommentResponseTo create(CommentRequestTo request) {
        Comment comment = commentMapper.toEntity(request);
        Comment saved = commentRepository.save(comment);
        return commentMapper.toDto(saved);
    }

    public CommentResponseTo update(Long id, CommentRequestTo request) {
        Comment comment = commentRepository.findById(id)
                .orElseThrow(() -> new NotFoundException("Comment not found with id: " + id));
        commentMapper.updateEntity(comment, request);
        comment.setModified(LocalDateTime.now());
        Comment updated = commentRepository.save(comment);
        return commentMapper.toDto(updated);
    }

    public void delete(Long id) {
        if (!commentRepository.existsById(id)) {
            throw new NotFoundException("Comment not found with id: " + id);
        }
        commentRepository.deleteById(id);
    }

    public List<CommentResponseTo> getByArticleId(Long articleId) {
        return commentRepository.findByArticleId(articleId).stream()
                .map(commentMapper::toDto)
                .collect(Collectors.toList());
    }
}

File: 351003/Pulish/app/services/comment_service.py
from app.dto.comment import CommentRequestTo, CommentResponseTo
from app.models.comment import Comment
from app.repository.inmemory import InMemoryRepository
from app.core.exceptions import NotFoundException


class CommentService:
    def __init__(self):
        self.repo = InMemoryRepository()

    def create(self, dto: CommentRequestTo) -> CommentResponseTo:
        comment = Comment(
            id=0,
            content=dto.content,
            topic_id=dto.topicId
        )
        saved = self.repo.create(comment)
        return self._to_response(saved)

    def find_all(self):
        return [self._to_response(c) for c in self.repo.find_all()]

    def find_by_id(self, id: int):
        comment = self.repo.find_by_id(id)
        if not comment:
            raise NotFoundException("Comment not found", 40404)
        return self._to_response(comment)

    def update(self, id: int, dto: CommentRequestTo):
        comment = self.repo.find_by_id(id)
        if not comment:
            raise NotFoundException("Comment not found", 40404)
        comment.content = dto.content
        comment.topic_id = dto.topicId
        return self._to_response(comment)

    def delete(self, id: int):
        comment = self.repo.find_by_id(id)
        if not comment:
            raise NotFoundException("Comment not found", 40401)
        self.repo.delete(id)

    def _to_response(self, comment: Comment) -> CommentResponseTo:
        return CommentResponseTo(
            id=comment.id,
            content=comment.content,
            topicId=comment.topic_id
        )
